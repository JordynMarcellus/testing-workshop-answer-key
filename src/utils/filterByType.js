//TODO: past jord note -- don't pass in an empty array in the final code. this is a subtle bug that will show why it's good to test ðŸ¤“
export const filterByType = (arrayOfPokemon, type) =>
  arrayOfPokemon.reduce((filteredPokemonArray, pokemonAtArrayIndex) => {
    if (pokemonAtArrayIndex.types.includes(type)) {
      return [...filteredPokemonArray, pokemonAtArrayIndex];
    }
    return filteredPokemonArray;
  });
