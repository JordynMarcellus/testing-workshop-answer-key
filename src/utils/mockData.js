export const mockPikaData = {
  data: {
    name: "pikachu",
    types: [
      {
        type: {
          name: "electric",
        },
      },
    ],
    sprites: {
      front_default: "path/to/img",
    },
  },
};

export const mockGolemData = {
  data: {
    name: "golem",
    types: [
      {
        type: {
          name: "rock",
        },
      },
      {
        type: {
          name: "ground",
        },
      },
    ],
    sprites: {
      front_default: "path/to/img",
    },
  },
};
