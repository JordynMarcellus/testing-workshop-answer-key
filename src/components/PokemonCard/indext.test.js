import React from "react";
import { PokemonCard } from "./";
import { render, cleanup } from "react-testing-library";
import { mockPikaData, mockGolemData } from "../../utils/mockData";
import "jest-dom/extend-expect";

describe("Pokemon Card", () => {
  afterEach(cleanup);
  it("renders a pokemon card for a single type", () => {
    const { container, debug } = render(
      <PokemonCard pokemon={mockPikaData.data} />
    );
    expect(container.firstChild).toMatchSnapshot();
    expect(container).toHaveTextContent("pikachu");
    expect(container).toHaveTextContent("electric");
  });
  it("renders a pokemon card for multiple types", () => {
    const { container } = render(<PokemonCard pokemon={mockGolemData.data} />);
    expect(container.firstChild).toMatchSnapshot();
    expect(container).toHaveTextContent("golem");
    expect(container).toHaveTextContent("rock");
    expect(container).toHaveTextContent("ground");
  });
});
