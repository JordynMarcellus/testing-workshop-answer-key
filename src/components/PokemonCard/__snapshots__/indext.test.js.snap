// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokemon Card renders a pokemon card for a single type 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-width: 300px;
}

<article
  class="emotion-0 emotion-1"
>
  <h1>
    pikachu
  </h1>
  <ul>
    <span>
      electric
    </span>
  </ul>
  <img
    alt="My chonky fave pikachu"
    src="path/to/img"
    title="My chonky fave pikachu"
  />
</article>
`;

exports[`Pokemon Card renders a pokemon card for multiple types 1`] = `
.emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  max-width: 300px;
}

<article
  class="emotion-0 emotion-1"
>
  <h1>
    golem
  </h1>
  <ul>
    <span>
      rock
    </span>
    <span>
      ground
    </span>
  </ul>
  <img
    alt="My chonky fave golem"
    src="path/to/img"
    title="My chonky fave golem"
  />
</article>
`;
